### PR TITLE
Dynamic ellipse

### PR DESCRIPTION
**Pull request by gotenxds**
performance depends on the granularity parameter.

2 changes per sec:

granularity = 0.03 - up to 700 ellipses with 30FPS.
granularity = 0.09 - up to 1500 ellipses with 30FPS.

### CODE REVIEW

File: src/angular-cesium/components/ac-layer/ac-layer.component.ts [hunk 1]
```diff
@@ -41,7 +42,7 @@ export class AcLayerComponent implements OnInit, OnChanges , AfterContentInit {
         this.entityName = acForArr[1];
 
         this.observable.merge(this._updateStream).subscribe((notification) => {
-            this._computationCache.clear()
+            this._computationCache.clear();
```
Comment (eitanfr) [inline src/angular-cesium/components/ac-layer/ac-layer.component.ts]: add `DynamicEllipseDrawerService` to `_drawerList`

File: src/app/tracks-layer/tracks-layer.component.ts [hunk 2]
```diff
@@ -41,6 +41,9 @@ export class TracksLayerComponent implements OnInit {
                     2000);
             });
         })
+
+        // this.tracks$ = Observable.from([{entity:Cesium.Cartesian3.fromRadians(Math.random(), Math.random()), actionType:ActionType.ADD_UPDATE, id:1}]);
```
Comment (eitanfr) [inline src/app/tracks-layer/tracks-layer.component.ts]: delete this line

File: src/angular-cesium/services/ellipse-drawer/dynamic-ellipse-drawer.service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            loop: true,
+            width: cesiumProps.width,
+            material: cesiumProps.material
+        };
+
+        return super.add(polylinedEllipse);
+    }
+
+    update(ellipse: any, cesiumProps: Object): any {
+        ellipse.positions = DynamicEllipseDrawerService.generatePositions(cesiumProps);
```
Comment (eitanfr) [inline src/angular-cesium/services/ellipse-drawer/dynamic-ellipse-drawer.service.ts]: what if `width` has changed ?or any other parameter except positions...

File: src/app/tracks-layer/tracks-layer.component.ts (hunk 2 above)
Comment (eitanfr) [inline src/app/tracks-layer/tracks-layer.component.ts]: I tried to run the code and got cesium length exception!
fix it. 
hit: merge from the master it should resolve it

**Comment by gotenxds** [comment]
@eitanfr Fixed.

**Pull request closed without merging**